packages/hawtio/src/plugins/camel/routes-service.ts

```diff
@@ -205,6 +256,102 @@ class RoutesService {
+    if (!routesNode) {
+      throw new Error('No route/routes node provided')
+    }
+
+    let xml = null
+    let mbeanName = routesNode.getProperty(contextNodeType)
+
+    try {
+      // if there is no mbean name then it's route node else it's routes folder
+      if (!mbeanName) {
+        mbeanName = routesNode.objectName ?? ''
+        xml = await jolokiaService.execute(mbeanName, 'dumpRouteStatsAsXml', [true, true])
+      } else {
+        xml = await jolokiaService.execute(mbeanName, 'dumpRoutesStatsAsXml', [true, true])
+      }
+    } catch (error) {
+      throw new Error('Failed to dump routes stats from mbean: ' + mbeanName + error)
+    }
+    if (!xml) {
+      throw new Error('Failed to extract any xml from mbean: ' + mbeanName)
+    }
+    return xml as string
+  }
+  processRoutesStats(statsXml: string): RouteStats[] {
+    const doc: XMLDocument = parseXML(statsXml)
+    const routesStats: RouteStats[] = []
+    const allRoutes = doc.getElementsByTagName('routeStat')
+    for (const route of allRoutes) {
+      routesStats.push(this.createRouteStats(route))
+    }
+    return routesStats
+  }
+
+  createProcessorStats(routeid: string, pDoc: Element): ProcessorStats {
+    return {
+      id: pDoc.id,
+      exchangesCompleted: parseInt(pDoc.getAttribute('exchangesCompleted') ?? '0'),
+      deltaProcessingTime: parseInt(pDoc.getAttribute('deltaProcessingTime') ?? '0'),
+      exchangeInflight: parseInt(pDoc.getAttribute('exchangeInflight') ?? '0'),
+      externalRedeliveries: parseInt(pDoc.getAttribute('externalRedeliveries') ?? '0'),
+      firstExchangeCompleteTimestamp: parseInt(pDoc.getAttribute('firstExchangeCompleteTimestamp') ?? '0'),
+      firstExchangeCompletedExchangeId: parseInt(pDoc.getAttribute('firstExchangeCompletedExchangeId') ?? '0'),
+      state: pDoc.getAttribute('state'),
+      index: parseInt(pDoc.getAttribute('index') ?? '0'),
+      lastExchangeCompleteTimestamp: parseInt(pDoc.getAttribute('lastExchangeCompleteTimestamp') ?? '0'),
+      maxProcessingTime: parseInt(pDoc.getAttribute('maxProcessingTime') ?? '0'),
+      meanProcessingTime: parseInt(pDoc.getAttribute('meanProcessingTime') ?? '0'),
+      minProcessingTime: parseInt(pDoc.getAttribute('minProcessingTime') ?? '0'),
+      redeliveries: parseInt(pDoc.getAttribute('redeliveries') ?? '0'),
+      resetTimestamp: parseInt(pDoc.getAttribute('resetTimestamp') ?? '0'),
+      sourceLineNumber: parseInt(pDoc.getAttribute('sourceLineNumber') ?? '0'),
+      startTimestamp: parseInt(pDoc.getAttribute('startTimestamp') ?? '0'),
+    }
+  }
+
+  createRouteStats(pDoc: Element): RouteStats {
+    const procStats = pDoc.getElementsByTagName('processorStat')
+    const processorStats: ProcessorStats[] = []
+    for (const pStat of procStats) {
+      processorStats.push(this.createProcessorStats(pDoc.id, pStat))
+    }
+
+    return {
+      id: pDoc.id,
+      deltaProcessingTime: parseInt(pDoc.getAttribute('deltaProcessingTime') ?? '0'),
```

Comment:
Since these parseInt functions are all the same, can they be replaced with a single shorter function?

eg.
```
parseAttr(pDoc: any, name: string) {
  pDoc.getAttribute(name) ....
}
```